reactor-netty-core/src/main/java/reactor/netty/resources/DefaultPooledConnectionProvider.java

```diff
@@ -127,7 +127,10 @@ static final class DisposableAcquire
 		// Deliberately suppress "NullAway"
 		// This is a lazy initialization
 		PooledRef<PooledConnection> pooledRef;
-		@Nullable Subscription subscription;
+		@SuppressWarnings("NullAway")
+		// Deliberately suppress "NullAway"
+		// This is a lazy initialization
+		Subscription subscription;
```

Comment:
I think it's easier to follow if the comment is above the relevant piece of code, so here's a suggestion of inverting the existing pattern. Also, a suggestion to explain why we don't expect an NPE instead of just mentioning lazy initialization. The deliberate intention is probably appearent from the context and the use of `SuppressWarnings("NullAway")` so I suggested removing that line.
```suggestion
		// Never null when accessed - only via dispose()
		// which is registered into sink.onCancel() callback.
		// See onSubscribe(Subscription).
		@SuppressWarnings("NullAway")
		Subscription subscription;
```

Reply:
fixed